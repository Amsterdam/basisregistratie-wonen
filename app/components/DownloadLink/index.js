import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@datapunt/asc-ui';

const DownloadLink = ({ name, label, onClick, disabled, ...rest }) => (
  <Link
    href={null}
    download={name}
    onClick={onClick}
    tabIndex={disabled ? -1 : 0}
    disabled={disabled}
    size="small"
    {...rest}
  >
    {label}
  </Link>
);

DownloadLink.defaultProps = {
  disabled: null,
  label: 'CSV',
  onClick: null,
};

DownloadLink.propTypes = {
  disabled: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default DownloadLink;
