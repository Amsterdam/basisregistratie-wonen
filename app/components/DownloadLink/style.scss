*[disabled] {
  background-color: #e6e6e6 !important;
  color: #767676 !important;
}
