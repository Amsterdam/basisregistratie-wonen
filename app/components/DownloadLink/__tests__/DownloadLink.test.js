import React from 'react';
import { render, cleanup } from 'react-testing-library';
import DownloadLink from '..';

describe('DownloadLink', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container, rerender } = render(<DownloadLink name="filename.csv" />);

    expect(container.firstChild).toMatchSnapshot();

    rerender(<DownloadLink name="filename.csv" disabled />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a value for the `download` attribute', () => {
    const { container } = render(<DownloadLink name="filename.csv" />);
    expect(container.firstChild.getAttribute('download')).toBe('filename.csv');
  });

  it('should render an achor element', () => {
    const { container } = render(<DownloadLink name="filename.csv" />);
    expect(container.firstChild).toBeInstanceOf(HTMLAnchorElement);
  });

  it('should have a negative tab-index when disabled', () => {
    const { container, rerender } = render(<DownloadLink name="filename.csv" />);

    expect(container.firstChild.tabIndex).toEqual(0);

    rerender(<DownloadLink name="filename.csv" disabled />);

    expect(container.firstChild.tabIndex).toEqual(-1);
  });
});
