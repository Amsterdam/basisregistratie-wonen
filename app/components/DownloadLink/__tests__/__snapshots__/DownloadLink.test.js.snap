// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadLink matches the snapshot 1`] = `
<a
  class="sc-gZMcBi eDfGXo sc-EHOje fmTxUw"
  download="filename.csv"
  tabindex="0"
>
  CSV
</a>
`;

exports[`DownloadLink matches the snapshot 2`] = `
<a
  class="sc-gZMcBi eDfGXo sc-EHOje fmTxUw"
  disabled=""
  download="filename.csv"
  tabindex="-1"
>
  CSV
</a>
`;
